galley/pkg/authplugins/google/google.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package google is a Galley auth plugin that uses Google application
+// default credentials.
+package google
+
+import (
+	"context"
+
+	"golang.org/x/oauth2/google"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/oauth"
+
+	"istio.io/istio/galley/pkg/authplugin"
+)
+
+type dcFn func(context.Context, ...string) (*google.Credentials, error)
+
+var findDC dcFn
+
+func init() {
+	findDC = google.FindDefaultCredentials
+}
+
+func returnAuth(config map[string]string) ([]grpc.DialOption, error) {
```

Comment:
It looks like we don't have a need for config for the time being. Should we remove that?

Reply:
Correct, it is not being used. The main reason we would want it is if we wanted to specify a service account file through config, instead of setting the `GOOGLE_APPLICATION_CREDENTIALS` env var.
Other auth plugins might want some sort of parameters as well.